src/testing.rs

```diff
@@ -12,7 +12,8 @@ use super::*;
 pub fn decode(encoding: &'static Encoding, bytes: &[u8], expect: &str) {
     let mut vec = Vec::with_capacity(bytes.len() + 32);
     let mut string = String::with_capacity(expect.len() + 32);
-    for i in 0usize..32usize {
+    let range = if cfg!(miri) { 0usize..4usize } else { 0usize..32usize };
```

Comment:
Would this be too slow if it went to `8usize`? That would end up testing all the ALU alignment possibilities.

Reply:
Hm, I can try that. Does the same apply to `encode`?

Reply:
Also what's "ALU alignment"?

Reply:
Going up to 8 is definitely not feasible with Stacked Borrows, then even my 16GB of RAM are insufficient.

But without Stacked Borrows it should be fine.

Reply:
Let's keep it as 4, then. By ALU alignment I meant the alignment of an `usize` and trying all byte alignments smaller than that.